test/integration/puppetlabs/puppetserver/testutils.clj

```diff
@@ -375,6 +375,17 @@
   ([puppet-conf-file dest-dir]
    (with-puppet-conf-fixture {"puppet.conf" puppet-conf-file} dest-dir)))
 
+(defmacro with-stub-puppet-conf
+  "Provides similar functionality of `with-puppet-conf` but uses a default
```

Comment:
I was a little confused by your comments on the other PR - should this reference `with-puppet-conf-files` instead?

Reply:
No, there's two ways in clojure to do the equivalent of Ruby's passing in a block. Define a function that takes another function, or define a macro that takes forms.

Clojure's test library has equivalents to Ruby RSpec's before, after, and around hooks, (these are the `use-fixtures` mentioned below) and they use the former approach, they take functions which are then passed functions (this is what `with-puppet-conf-fixture` and transitively `with-puppet-conf` are geared towards).

Inside of actual deftests this project tends to go with the latter, macro approach to "block passing".

So this macro does something most similar to what `with-puppet-conf` does, but is implemented in a way to be used outside of the "rspec around-style hook" that `with-puppet-conf` is geared towards.

(Note you can actually mix and match those approaches (iirc, code-manager uses function passing for everything), but I was just trying to follow _what seemed idiomatic for this project_)

Reply:
gotcha, thank you for explaining!